File: typescript/tests/Sonde.test.ts
import { Direction } from '../src/Direction'
import { Helicoptere } from '../src/Helicoptere'
import { ModuleMartien2D } from '../src/ModuleMartien2D'
import { ModuleMartien3D } from '../src/ModuleMartien3D'
import { MouvementNord2D } from '../src/MouvementNord2D'
import { Position } from '../src/Position'
import { Rover } from '../src/Rover'
import { Sonde } from '../src/Sonde'

describe('Sonde', () => {
  it('Le rover doit être récupéré', () => {
    // GIVEN
    const nord = new Direction(new MouvementNord2D());
    const perseverance: ModuleMartien2D = new Rover(nord, new Position(4, 4, 0));
    const sonde: Sonde = new Sonde();

    // WHEN
    sonde.preparerRecuperation(perseverance);

    // THEN
    expect(perseverance.isPretPourRecuperation()).toBeTruthy();
    expect(new Position(4, 4, 0)).toStrictEqual(perseverance.getPosition())
  })

  it('L’hélicoptère doit être récupéré', () => {
    // GIVEN
    const nord = new Direction(new MouvementNord2D());
    const ingenuity: ModuleMartien3D = new Helicoptere(nord, new Position(4, 4, 50));
    const sonde: Sonde = new Sonde();

    // WHEN
    sonde.preparerRecuperation(ingenuity);

    // THEN
    expect(ingenuity.isPretPourRecuperation()).toBeTruthy();
    expect(new Position(4, 4, 0)).toStrictEqual(ingenuity.getPosition())
  })
})

File: java/mission-to-mars-solid/src/test/java/mars/SondeTest.java
package mars;

import static org.junit.Assert.assertTrue;
import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class SondeTest {

	@Test
	void roverDoitEtreRecupere() {
		// Arrange
		Sonde sut = new Sonde();
		ModuleMartien2D perseverance = new Rover(Direction.NORD, new Position(4, 4, 0));

		// Act
		sut.preparerRecuperation(perseverance);

		// Assert
		assertTrue(perseverance.isPretPourRecuperation());
		assertEquals(new Position(4, 4, 0), perseverance.getPosition());
	}

	@Test
	void helicoptereDoitEtreRecupere() {
		// Arrange
		Sonde sut = new Sonde();
		ModuleMartien3D ingenuity = new Helicoptere(Direction.NORD, new Position(4, 4, 50));

		// Act
		sut.preparerRecuperation(ingenuity);

		// Assert
		assertTrue(ingenuity.isPretPourRecuperation());
		assertEquals(new Position(4, 4, 0), ingenuity.getPosition());
	}
}
